main.py

```diff
@@ -3,14 +3,26 @@
 """
 
 import os
+import sys
+import time
 from datetime import datetime
 
-from documentcloud.addon import AddOn
+from documentcloud.addon import AddOn, SoftTimeOutAddOn
 from documentcloud.toolbox import requests_retry_session
 from requests.exceptions import RequestException
 
 
-class Filecoin(AddOn):
+class Filecoin(SoftTimeOutAddOn):
+    soft_time_limit = 240
```

Comment:
Can probably increase this to 5 or 10 minutes - we have a 30 minute hard timeout